core-bundle/src/Routing/Content/PageProviderInterface.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Content;
+
+use Contao\PageModel;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Route;
+
+/**
+ * A page provider provides information about a page type.
+ * The page provider knows what route for a page of its type looks like, e.g. if
+ * a particular route has a different URL suffix than configured in the root page.
+ */
+interface PageProviderInterface
```

Comment:
I agree. I don't like `PageType` because that would be an _instance_ of something, not a service. I also want to get away from the existing page types / find a name so these are not confused with each other.

Reply:
The summary of what it does in the description is too long. Try shortening what it does to a simple sentence, that should help with naming :)

Reply:
As discussed in Slack:

```php
namespace Contao\CoreBundle\Routing\Content;

interface PageTypeInterface
{
    public const ATTRIBUTE = '_page_provider';

    public function getName(): string;

    public function getRoute(PageModel $pageModel, $content = null, Request $request = null): Route;

    public function getUrlSuffixes(): array;

    public function supportsContentComposition(PageModel $pageModel = null): bool;
}
```